integration/container/exec_test.go

```diff
@@ -117,3 +117,46 @@ func TestExec(t *testing.T) {
 	assert.Assert(t, is.Contains(out, "PWD=/tmp"), "exec command not running in expected /tmp working directory")
 	assert.Assert(t, is.Contains(out, "FOO=BAR"), "exec command not running with expected environment variable FOO")
 }
+
+func TestExecKill(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.35"), "broken in earlier versions")
+	skip.If(t, testEnv.OSType == "windows", "FIXME. Probably needs to wait for container to be in running state.")
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(t, ctx, client, container.WithTty(true), container.WithWorkingDir("/root"))
+
+	// Assume this test ends within 10 seconds
+	id, err := client.ContainerExecCreate(ctx, cID,
+		types.ExecConfig{
+			WorkingDir:   "/tmp",
+			Env:          strslice.StrSlice([]string{"FOO=BAR"}),
+			AttachStdout: true,
+			Cmd:          strslice.StrSlice([]string{"sh", "-c", "for i in {1..10}; do sleep 1; done"}),
+		},
+	)
+	assert.NilError(t, err)
+
+	resp, err := client.ContainerExecAttach(ctx, id.ID, types.ExecStartCheck{})
+	assert.NilError(t, err)
+	defer resp.Close()
+
+	_, err = client.ContainerExecInspect(ctx, id.ID)
+	assert.NilError(t, err)
+
+	err = client.ContainerExecKill(ctx, id.ID, types.ExecKillOptions{})
+	assert.NilError(t, err)
+
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
why?

Reply:
I'd like to have your opinion about the behavior of the exec kill feature. Currently, it just sends a signal to the process and doesn't remove the instance from `execCommands`. That's why I assumed it might not finish just after calling the method. If we want to make the kill action with a synchronized manner, we have to wait for the process termination in `ContainerExecKill` to remove the exec instance, but it requires timeout in its option. On the otherhand, it we remove the exec intance just after sending a signal to the process, we can make it synchronized easily without timeout, but have possibility of resource leakage.
Which do you think is better?

Reply:
To consider non-force cases (TERM` signal), I think `ContainerExecKill` should just send a signal to the running process and the caller should handle the result just like the `kill` command of Linux.